climada/hazard/base.py

```diff
@@ -157,49 +158,88 @@ class Hazard():
     """Name of the variables that aren't need to compute the impact. Types:
     scalar, string, list, 1dim np.array of size num_events."""
 
-    def __init__(self, haz_type='', pool=None):
-        """
-        Initialize values.
+    def __init__(self,
+                 haz_type: str,
```

Comment:
This is the only (?) major change of the interface so far. Here, I'm requiring users to specify the hazard type. I opted for this because I feel like the hazard type is a very important bit of information. I could give it an empty string as default value, but then I would need to move the argument behind the parameters without default types. That would make it look "less important" to me.

@chahank, what do you think about this?

Reply:
Actually, to be consistent with the previous API, all attributes must have the same defaults as before. I think changing the defaults (and removing some) is a major change, and should not be done lightly. I would, although it is still sub-optimal, for this update, just keep all the defaults.

Reply:
Then I have a follow-up issue: The `haz_type` is actually an attribute of the `Tag`, not the Hazard. So for keeping the previous defaults, we could simply remove the `haz_type` from the argument list. It will then become part of the `tag_kwargs`. However, that obscures this very important variable. What to do?

Reply:
I do not understand. Currently, the default is `haz_type=''`, and it is a variable of the `__init__`... why would that be a problem now? You can just leave it as it is right?

Reply:
This is the [same default as in `Tag`](https://github.com/CLIMADA-project/climada_python/blob/main/climada/hazard/tag.py#L42). Such double-defaults are dangerous because the default in `Hazard` would ignore any changes to the default in `Tag`.

Another viable option I see is to have users insert a proper `Tag` object as optional parameter. If none is passed, we initialize a default one (this also default-inits the `haz_type`).

Reply:
Yes, but this is a problem not to be solved here. The whole `Tag` needs a complete overhaul. This is also in the pipeline ;).

Reply:
See 06c40654146bc41134af06a99b673f7023bf3e32